File: BlackJack.ViewModels/HistoryViews/DetailsOfGameHistoryView.cs
using BlackJack.ViewModels.EnumViews;
using System;
using System.Collections.Generic;

namespace BlackJack.ViewModels.HistoryViews
{
    public class DetailsOfGameHistoryView
    {
        public GameDetailsOfGameHistoryView Game { get; set; }
        public PlayerAndBotStepsDetailsOfGameHistoryView PlayerAndBotSteps { get; set; }
    }

    public class GameDetailsOfGameHistoryView
    {
        public Guid Id { get; set; }
        public string WonName { get; set; }
        public GameStateTypeEnumView GameState { get; set; }
        public PlayerDetailsOfGameHistoryView Player { get; set; }
    }

    public class PlayerDetailsOfGameHistoryView
    {
        public string Id { get; set; }
        public string UserName { get; set; }
        public decimal Balance { get; set; }
        public decimal Bet { get; set; }
    }
    
    public class PlayerAndBotStepsDetailsOfGameHistoryView
    {
        public List<StepPlayerAndBotStepsDetailsOfGameHistoryViewItem> Steps { get; set; }

        public PlayerAndBotStepsDetailsOfGameHistoryView()
        {
            Steps = new List<StepPlayerAndBotStepsDetailsOfGameHistoryViewItem>();
        }
    }

    public class StepPlayerAndBotStepsDetailsOfGameHistoryViewItem
    {
        public List<CardPlayerAndBotStepsDetailsOfGameHistoryViewItem> Cards { get; set; }

        public StepPlayerAndBotStepsDetailsOfGameHistoryViewItem()
        {
            Cards = new List<CardPlayerAndBotStepsDetailsOfGameHistoryViewItem>();
        }
    }

    public class CardPlayerAndBotStepsDetailsOfGameHistoryViewItem
    {
        public Guid Id { get; set; }
        public SuiteTypeEnumView Suite { get; set; }
        public RankTypeEnumView Rank { get; set; }
        public PlayerCardPlayerAndBotStepsDetailsOfGameHistoryView Player { get; set; }
        public GameCardPlayerAndBotStepsDetailsOfGameHistoryView Game { get; set; }
        public BotCardPlayerAndBotStepsDetailsOfGameHistoryView Bot { get; set; }
    }
    
    public class PlayerCardPlayerAndBotStepsDetailsOfGameHistoryView
    {
        public string Id { get; set; }
        public string UserName { get; set; }
        public decimal Balance { get; set; }
        public decimal Bet { get; set; }
    }

    public class GameCardPlayerAndBotStepsDetailsOfGameHistoryView
    {
        public Guid Id { get; set; }
        public string WonName { get; set; }
        public GameStateTypeEnumView GameState { get; set; }
    }

    public class BotCardPlayerAndBotStepsDetailsOfGameHistoryView
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public decimal Balance { get; set; }
        public decimal Bet { get; set; }
    }
}

File: BlackJack.WEB/ClientApp/src/app/shared/entities/history.views/details-of-game.history.view.ts
import { GameStateType } from "../../enums/game-state-type";
import { SuiteType } from "../../enums/suite-type";
import { RankType } from "../../enums/rank-type";

export class DetailsOfGameHistoryView {
  game?: GameDetailsOfGameHistoryView;
  playerAndBotSteps?: PlayerAndBotStepsDetailsOfGameHistoryView;

  constructor() {
    this.game = new GameDetailsOfGameHistoryView();
    this.playerAndBotSteps = new PlayerAndBotStepsDetailsOfGameHistoryView();
  }
}

export class PlayerAndBotStepsDetailsOfGameHistoryView {
  steps?: Array<StepPlayerAndBotStepsDetailsOfGameHistoryViewItem>;

  constructor() {
    this.steps = new Array<StepPlayerAndBotStepsDetailsOfGameHistoryViewItem>();
  }
}

export class StepPlayerAndBotStepsDetailsOfGameHistoryViewItem {
  cards?: Array<CardPlayerAndBotStepsDetailsOfGameHistoryView>;

  constructor() {
    this.cards = new Array<CardPlayerAndBotStepsDetailsOfGameHistoryView>();
  }
}

export class CardPlayerAndBotStepsDetailsOfGameHistoryView {
  id?: string;
  suite?: SuiteType;
  rank?: RankType;
  player?: PlayerCardPlayerAndBotStepsDetailsOfGameHistoryView;
  game?: GameCardPlayerAndBotStepsDetailsOfGameHistoryView;
  bot?: BotCardPlayerAndBotStepsDetailsOfGameHistoryView;

  constructor() {
    this.player = new PlayerCardPlayerAndBotStepsDetailsOfGameHistoryView();
    this.game = new GameCardPlayerAndBotStepsDetailsOfGameHistoryView();
    this.bot = new BotCardPlayerAndBotStepsDetailsOfGameHistoryView();
  }
}

export class PlayerCardPlayerAndBotStepsDetailsOfGameHistoryView {
  userName?: string;
  id?: string;
  balance?: number;
  bet?: number;
}

export class GameCardPlayerAndBotStepsDetailsOfGameHistoryView {
  id?: string;
  wonName?: string;
  gameState?: GameStateType;
}

export class BotCardPlayerAndBotStepsDetailsOfGameHistoryView {
  id?: string;
  name?: string;
  balance?: number;
  bet?: number;
}

export class GameDetailsOfGameHistoryView {
  id?: string;
  wonName?: string;
  gameState?: GameStateType;
  player?: PlayerDetailsOfGameHistoryView;

  constructor() {
    this.player = new PlayerDetailsOfGameHistoryView();
  }
}

export class PlayerDetailsOfGameHistoryView {
  userName?: string;
  id?: string;
  balance?: number;
  bet?: number;
}
